test/framework/machine_helpers.go

```diff
@@ -21,32 +21,109 @@ import (
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
-	"github.com/pkg/errors"
 
+	"github.com/pkg/errors"
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
 )
 
-// WaitForMachinesToBeUpgradedInput is the input for WaitForMachinesToBeUpgraded.
-type WaitForMachinesToBeUpgradedInput struct {
-	Lister                   Lister
-	Cluster                  *clusterv1.Cluster
-	KubernetesUpgradeVersion string
-	MachineCount             int
+type GetMachinesByClusterInput struct {
```

Comment:
My personal preference goes to avoid shared types across helpers Input struct because I don't like the nesting it introduces when used and for limiting the blaste of potential breaking changes in future